### ISSUE TITLE
Dataset - View Dataset Versions button does not appear on page load

### ISSUE THREAD

**Issue by sekmiller** [IQSS/dataverse#2782]
One of the things we did to improve performance was to not load the versions onto the Versions Tab until the user actually went to the Versions tab. A side effect of that was that the View Dataset Versions button was not available on page load because its rendering is dependent on the presence of multiple versions, either published or one published and one draft if the logged in user has the permission to view drafts. In order to return the button we probably have to lose those performance gains since the View Versions button has a drop-down list of versions which has to be populated by processing all of the dataset's versions.

**Comment by eaquigley** [comment]
Additionally, the versions tab says, "no records found" until it starts loading (which takes about a second I believe) and can be concerning to a user who has a dataset with a lot of versions as it comes across as data is missing.

**Comment by mheppler** [comment]
pdurbin commented on Apr 26 on issue #3092

> "No records found." To navigate to the draft you have to leave the dataset page.

![screen shot 2016-04-26 at 3 12 03 pm](https://cloud.githubusercontent.com/assets/21006/14830969/6d426a90-0bc1-11e6-8f5e-750ff18fd106.png)

**Comment by mheppler** [comment]
The issue with the Version table on a deaccessioned dataset reading "No records found" has been fixed in #1267.

**Comment by mheppler** [comment]
I am adding this to 4.6.1 to be included as part of the development with #3241, as [recommended by the UX Team](https://docs.google.com/document/d/1mNNRqRqIW92LXock332sIA-uXh-2agOpoyLLLKwXUA0/edit) based upon user feedback in usability testing of 4.6.

**Comment by mheppler** [comment]
Adding these recommendations on the file pg to this issue.

- When there are no tags on a file, we need an indicator that tags are a possibility. Show “No tags” in the tags area if there are no tags.
- Need a success message after restricting access to a file.

Recommended by the UX Team based upon user feedback in usability testing of 4.6.

**Comment by djbrooke** [comment]
We discussed this in the planning meeting this afternoon. We identified the following items:

- [ ] Provide the list of links as part of the page load
- [ ] Determine the design of these (dropdown, button, list, other?)
- [ ] Implement the links on the page

**Comment by landreev** [comment]
@mheppler , I created a branch - 2782-show-versions-on-dataset-page - and added code to provide a list of other versions available (that the user is allowed to see). And added the links to the page; as a simple list: 
![screen shot 2017-01-12 at 7 02 24 pm](https://cloud.githubusercontent.com/assets/737558/21913397/3e4fefbc-d8fa-11e6-9101-ac417024c570.png)

please modify as you see fit - put them in a pulldown menu and/or style the links pretty, etc.

**Comment by mheppler** [comment]
Much appreciated, @landreev.

**Comment by mheppler** [comment]
Pushing through new UI.

![screen shot 2017-01-19 at 12 23 03 pm](https://cloud.githubusercontent.com/assets/687227/22117455/1c0908c4-de42-11e6-8fe1-43dca3c2405a.png)

**Comment by pdurbin** [comment]
@mheppler I'm running the code in pull request #3580 (I'm on 077bd73) and there are a couple consistency things I'm wondering about.

First, on the "Versions" tab we show 1.0 and 2.0 but in the new widget we show v1.0 and v2.0 with a "v". Should these be the same?

Second, when 1.0 has been deaccessioned, as in the example screenshot below, "1.0" is not clickable in the "Versions" tab but it is clickable in the new widget. It seems inconsistent.

<img width="1206" alt="screen shot 2017-01-19 at 12 45 32 pm" src="https://cloud.githubusercontent.com/assets/21006/22118631/8f2f82fc-de46-11e6-84c3-d8078ae18d7c.png">

**Comment by mheppler** [comment]
@pdurbin -- it was decided these design questions would be pushed until we discuss file-versions. The deaccession link I will work to fix now.

**Comment by pdurbin** [comment]
@sekmiller said he's code review this one. Assigning.

**Comment by mheppler** [comment]
Thanks, @pdurbin and @sekmiller.

**Comment by mheppler** [comment]
Merged the latest and greatest from `develop`. 7fec06f472b3e8fbb72a756a2b3fb6d2b5f4e88a

**Comment by kcondon** [comment]
This works fine but may need some Design attention for cases where there are many versions in a dataset.

**Comment by djbrooke** [comment]
@kcondon - can you link up an example of a case of many datasets? Thanks!

**Comment by kcondon** [comment]
doi:10.7910/DVN/VIE1H has 306 versions, the largest.

Version count stats:
datasets with versions >1: 8895
datasets with versions >5: 193
datasets with versions >10: 25
datasets with versions >15: 8
datasets with versions >20: 5
datasets with versions > 25: 1

**Comment by djbrooke** [comment]
Pulling this out of the current sprint and out of 4.6.1. The complexity has increased, the value provided by this is minimal, and this even has the potential to provide a worse experience for datasets with many versions. 

We will transition to work defined on [the roadmap](http://dataverse.org/goals-roadmap-and-releases) instead of continuing to work on this issue at this time.

**Comment by pdurbin** [comment]
I think what would help unblock this issue is to get a screenshot of what the version labels look like when there are 25 versions of a dataset.

**Comment by pdurbin** [comment]
At https://groups.google.com/d/msg/dataverse-community/bGlCU2pbQpE/vAKtSAjxAwAJ @philippconzett from UiT, The Arctic University of Norway, wrote,

> Actually, there are some more features which would make the curator's job easier
> 
> 1. Include the version number in the submitting message, i.e. when a dataset is submitted for review, the message to the curator should state whether this is an entirely new dataset, or a new version of an existing dataset.

This was in the context of #3943 and I summarized all of his feedback (and some from @stevenmce ) at https://groups.google.com/d/msg/dataverse-community/bGlCU2pbQpE/10oSARygAQAJ

Here's the bit I wrote about version numbers:

> My second observation is that curators are potentially confused about whether they are being to asked to curate an entirely new dataset that has never been published (a combination of the labels "Unpublished" and "Draft" indicates this brand new, never been published status) versus a dataset that has already been published as 1.0 or whatever but now has a post-publication draft (the label "Draft" is present but the label "Unpublished" is absent). It's probably too subtle to expect people to think this hard about the presence or absence of the "Unpublished" label. For now, I hope this helps explain the existing labels. If this isn't documented in the User Guide please do open a GitHub issue. :) Anyway, your proposed solution is to show the version number to the curator because the curator knows that "1.1" means the dataset has already been published and that the author needs review of a post-publication draft. It's a good idea. There's a related GitHub issue at https://github.com/IQSS/dataverse/issues/2782 about bringing back a version indicator on the dataset page that was lost when we made some performance improvements and I think this would help a ton. This is probably out of scope for https://github.com/IQSS/dataverse/issues/3943 but I can at least look into adding the version number to the *notifications* that are sent.

It all feels on-topic enough to mention here.

**Comment by mheppler** [comment]
After discussing with @scolapasta and @djbrooke last week, it was decided to just to simply remove this button from the page.

**Comment by mheppler** [comment]
There was a commented out line in the User Guide referencing the button in the UI, which was also removed.
